torch_uncertainty/datamodules/cubic_regression.py

```diff
@@ -0,0 +1,119 @@
+from argparse import ArgumentParser
```

Comment:
If we keep the datamodule, it may be preferable to put it into a "toy" folder or file, for instance. However, I'm not entirely sure it is relevant to include it in the library, at least here. @alafage any thoughts? We could also put it into the utils folder? Another possibility is to implement it as a dataset, which makes more sense as we expect much more diversity on this side.

Depending on what we make of the datamodule, we could generalize it to polynomial expressions.

Reply:
I second the idea of implementing this as a dataset rather than a datamodule. After discussions with @o-laurent, we will add two new folders within `datasets/`: `classification/` and `regression/` to organize our datasets further. The cubic regression dataset would be included in the `datasets/regression/toy.py` file.

Reply:
For information, I've started writing the agreed modifications.